r/src/compute-exec.cpp

```diff
@@ -277,4 +278,65 @@ std::shared_ptr<compute::ExecNode> ExecNode_ReadFromRecordBatchReader(
   return MakeExecNodeOrStop("source", plan.get(), {}, options);
 }
 
+// just until we figure out the incantation to get a proper sinknode
```

Comment:
I think we would eventually want to do the same thing we do with exec plans which is to consume with pushgenerator (which is basically a producer/consumer queue).  The sink node consumer will push the batch into the queue and then a record batch reader will read the batches from the queue.  We'd also want to wire it up to backpressure like we have today.

Reply:
In fact, to reduce duplication, we may want to add a C++ method that takes in a Substrait plan, and returns a record batch reader.

Reply:
ARROW-15849